File: app/src/main/java/com/lttrung/dormitory/database/repositories/RoomRepositories.java
package com.lttrung.dormitory.database.repositories;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Single;

@Singleton
public interface RoomRepositories {
    Single<Type> fetchRooms();
    Single<Type> fetchRoomDetails();
}

File: app/src/main/java/com/lttrung/dormitory/database/repositories/RoomRepositories.kt
package com.lttrung.dormitory.database.repositories

import com.lttrung.dormitory.database.data.network.RoomNetwork
import com.lttrung.dormitory.database.data.network.models.Room
import io.reactivex.rxjava3.core.Single
import java.lang.reflect.Type
import javax.inject.Singleton

@Singleton
interface RoomRepositories {
    val network: RoomNetwork
    fun fetchRooms(roomTypeId: Int): Single<List<Room>>
    fun fetchRoomDetails(): Single<Type>
}